scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```
@@ -63,13 +63,15 @@ object Common {
 
     FileInputFormat
       .getInputPaths(conf)
-      .map { path =>
-        val fs = path.getFileSystem(conf)
-        val size = fs.globStatus(path)
-          .map(status => fs.getContentSummary(status.getPath).getLength)
-          .sum
-
-        path -> size
+      .flatMap { path =>
+        Try {
+          val fs = path.getFileSystem(conf)
+          val size = fs.globStatus(path)
```

we should still check for null so that we can have better error messages right? Actually this Try {} block is going to swallow exceptions quietly which doesn't seem great? The explicit check for null via the Option constructor seemed better -- if we do get an IOException here, wouldn't we want it to bubble up?

---

+1. If the NPE is the issue, let's handle it directly:
```scala
val size = fs.globStatus(path) match {
  case null => 0
  case items => items.map(...).sum
}
```

---

That said, I'm not sure you want to be failing the job here if you have an exception. Presumably we want to just log and give warnings that reducer estimation may be off and go ahead assuming they were empty?